diesel_codegen_new/src/util.rs

```diff
@@ -0,0 +1,128 @@
+use std::mem;
+use syn::*;
+
+use ast_builder::ty_ident;
+
+pub fn struct_ty(name: Ident, generics: &Generics) -> Ty {
+    let lifetimes = generics.lifetimes.iter().map(|lt| lt.lifetime.clone()).collect();
+    let ty_params = generics.ty_params.iter()
+        .map(|param| ty_ident(param.ident.clone()))
+        .collect();
+    let parameter_data = AngleBracketedParameterData {
+        lifetimes: lifetimes,
+        types: ty_params,
+        bindings: Vec::new(),
+    };
+    let parameters = PathParameters::AngleBracketed(parameter_data);
+    Ty::Path(None, Path {
+        global: false,
+        segments: vec![
+            PathSegment {
+                ident: name,
+                parameters: parameters,
+            },
+        ],
+    })
+}
+
+pub fn str_value_of_attr_with_name<'a>(
+    attrs: &'a [Attribute],
+    name: &str,
+) -> Option<&'a str> {
+    attr_with_name(attrs, name).map(|attr| str_value_of_attr(attr, name))
+}
+
+pub fn ident_value_of_attr_with_name<'a>(
+    attrs: &'a [Attribute],
+    name: &str,
+) -> Option<&'a Ident> {
+    attr_with_name(attrs, name).map(|attr| single_arg_value_of_attr(attr, name))
+}
+
+pub fn list_value_of_attr_with_name<'a>(
+    attrs: &'a [Attribute],
+    name: &str,
+) -> Option<Vec<&'a Ident>> {
+    attr_with_name(attrs, name).map(|attr| list_value_of_attr(attr, name))
+}
+
+pub fn attr_with_name<'a>(
+    attrs: &'a [Attribute],
+    name: &str,
+) -> Option<&'a Attribute> {
+    let ident_name = Ident::new(name);
+    attrs.into_iter().find(|attr| *attr_name(attr) == ident_name)
+}
+
+pub fn attr_name(attr: &Attribute) -> &Ident {
+    match attr.value {
```

Comment:
I added `attr.value.name() -> &str` in syn 0.7.1 and `attr.name() -> &str` in 0.8.5.